libfwupdplugin/fu-i2c-device.c

```diff
@@ -80,9 +80,33 @@ fu_i2c_device_probe(FuDevice *device, GError **error)
 
 	/* set the device file manually */
 	if (fu_udev_device_get_device_file(FU_UDEV_DEVICE(self)) == NULL) {
-		g_autofree gchar *devfile =
-		    g_strdup_printf("/dev/i2c-%u",
-				    (guint)fu_udev_device_get_number(FU_UDEV_DEVICE(self)));
+		const gchar *sysfs = fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(self));
+		g_auto(GStrv) tokens = fu_strsplit(sysfs, strlen(sysfs), "/", -1);
+		g_autofree gchar *devfile = NULL;
+		guint64 number = G_MAXUINT64;
+
+		for (guint i = 0; tokens[i] != NULL; i++) {
```

Comment:
isn't this going from root to tip? don't you want to go from tip to root?

Reply:
Yeah; it's root to tip.  But I had figured for an I2C device you _want_ the root bus.  Disagree?

Reply:
Maybe it's an angels on pinheads thing. I don't mind too much each way. I've got a megacommit which adds fake sysfs devices coming up (to test all this fu-udev class stuff I added) -- we can add a test for this condition there.

Reply:
> I've got a megacommit which adds fake sysfs devices coming up (to test all this fu-udev class stuff I added

Like we can actually get to the point someone can tar up /sys and we can emulate someone's udev devices?

Reply:
> Like we can actually get to the point someone can tar up /sys and we can emulate someone's udev devices?

Exactly that